patches/tModLoader/Terraria/ModLoader/GlobalItem.cs

```diff
@@ -276,10 +276,10 @@ public abstract class GlobalItem : ModType
 		}
 
 		/// <summary>
-		/// Allows you to determine whether a melee weapon can hit the given NPC when swung. Return true to allow hitting the target, return false to block the weapon from hitting the target, and return null to use the vanilla code for whether the target can be hit. Returns null by default.
+		/// Allows you to determine whether a melee weapon can hit the given NPC when swung. Return true to allow hitting the target, return false to block the weapon from hitting the target. Returns true by default.
 		/// </summary>
-		public virtual bool? CanHitNPC(Item item, Player player, NPC target) {
-			return null;
+		public virtual bool CanHitNPC(Item item, Player player, NPC target) {
+			return true;
```

Comment:
Why is this?

Reply:
Why should they not be normal booleans? If all of them return true, nothing happens. If one of them returns false, the NPC can't be hit.

Reply:
I'm actually not sure how to do this. In vanilla, the first things that are checked are things like if the player has the no critter harm item, if the npc is friendly, if it's the clothier, if it's invincible. Not if it is colliding. So I think I would have to use a goto  to skip the ifs? If null is returned, let the vanilla checks run, if false is returned, just skip over the npc, if true is returned, skip immediately to the collision check. Does that sound right?

Reply:
Isn't that how it was in the first place? Null is usually so you can tell to let vanilla do it's thing